brain/detail/circuit.h

```diff
@@ -214,20 +183,384 @@ class Circuit::Impl
 
     virtual Vector3fs getPositions(const GIDSet& gids) const = 0;
     virtual size_ts getMTypes(const GIDSet& gids) const = 0;
-    virtual Strings getMorphologyNames() const = 0;
+    virtual Strings getMorphologyTypeNames() const = 0;
     virtual size_ts getETypes(const GIDSet& gids) const = 0;
     virtual Strings getElectrophysiologyNames() const = 0;
     virtual Quaternionfs getRotations(const GIDSet& gids) const = 0;
     virtual Strings getMorphologyNames(const GIDSet& gids) const = 0;
 
     URI getMorphologyURI(const std::string& name) const
     {
-        URI uri(_morphologySource);
+        if (boost::filesystem::path(name).is_absolute())
+            return URI(name);
+
+        URI uri(getMorphologySource());
         uri.setPath(uri.getPath() + "/" + name + ".h5");
         return uri;
```

Comment:
I'll review this and change it myself. I will commit on top of your branch head.